### PR TITLE
Add xfinity-wifi

### PR DESCRIPTION
**Pull request by szhu**
#### Adding a new cask
- [X] Checked there aren’t open [pull requests](https://github.com/caskroom/homebrew-cask/pulls) for the same cask.
- [X] Checked there aren’t closed [issues](https://github.com/caskroom/homebrew-cask/issues) where that cask was already refused.
- [X] When naming the cask, followed the [token reference](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md).
- [X] Commit message includes cask’s name.
- [X] `brew cask audit --download {{cask_file}}` is error-free.
- [X] `brew cask style --fix {{cask_file}}` left no offenses.
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.

---

Can someone help me figure out why this isn't working? I can `curl` and `wget` the URL and it produces a file but it seems like no file is being placed in the `latest` folder.

### CODE REVIEW

File: Casks/xfinity-wifi.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+cask 'xfinity-wifi' do
+  version :latest
+  sha256 :no_check
+
+  url 'http://wifi.xfinity.com/download_mac.php'
+  name 'XFINITY WiFi'
+  homepage 'http://wifi.xfinity.com/'
+  license :gratis
+
+  pkg 'download_mac.php'
```
Comment (fanquake) [inline Casks/xfinity-wifi.rb]: pkg should be `XFINITY_WiFi.pkg

File: Casks/xfinity-wifi.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+cask 'xfinity-wifi' do
+  version :latest
+  sha256 :no_check
+
+  url 'http://wifi.xfinity.com/download_mac.php'
+  name 'XFINITY WiFi'
+  homepage 'http://wifi.xfinity.com/'
```
Comment (fanquake) [inline Casks/xfinity-wifi.rb]: Can you use the https link for the homepage `https://wifi.xfinity.com`

**Comment by szhu** [comment]
I tried that originally. It doesn't work. I don't think HBC/`curl` detects the HTTP filename but instead uses the one from the URL.

**Comment by jawshooah** [comment]
Needs an [`uninstall` stanza](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/uninstall.md#uninstall-key-pkgutil).

**Comment by szhu** [comment]
I will add one once the install part is fixed.

**Comment by jawshooah** [comment]
For now we'll have to make do with the `preflight` solution used in [`fuse.rb`](https://github.com/caskroom/homebrew-cask/blob/312ae841f1f1b2ec07f4d88b7dfdd7fbdf8d4f94/Casks/fuse.rb):

```rb
cask 'xfinity-wifi' do
  version :latest
  sha256 :no_check

  url 'http://wifi.xfinity.com/download_mac.php'
  name 'XFINITY WiFi'
  homepage 'https://wifi.xfinity.com/'
  license :gratis

  container type: :pkg

  # This is a horrible hack to force the file extension.  The
  # backend code should be fixed so that this is not needed.
  preflight do
    system '/bin/mv', '--', staged_path.join('download_mac.php'), staged_path.join('XFINITY_WiFi.pkg')
  end

  pkg 'XFINITY_WiFi.pkg'
end
```

**Comment by jawshooah** [comment]
Also looks like this is what's needed for `uninstall`:

```rb
  uninstall pkgutil:   'com.smithmicro.netwise.osx.comcast.installer',
            launchctl: 'com.smithmicro.netwise.osx.helper',
            quit:      [
                         'com.smithmicro.netwise.HotspotAuthenticatorOSX',
                         'com.smithmicro.netwise.osx.comcast',
                         'com.smithmicro.netwise.osx.comcast.help',
                       ]
```

**Comment by jawshooah** [comment]
Thank you for the contribution. It needed some fixes, so they were made in 153a0161d2ed5f124544fbc32195f83636dcbe19. Your contribution is still included (and still credited to you), with the appropriate modifications. Please feel free to ask about any of the changes.

**Comment by szhu** [comment]
Wow, thanks! That commit should have been credited to you :)

**Pull request closed without merging**